hoomd/mpcd/pytest/test_collide.py

```diff
@@ -130,3 +145,56 @@ def test_run(self, small_snap, simulation_factory, cls, init_args):
             period=1, embedded_particles=hoomd.filter.All(), **init_args
         )
         sim.run(1)
+
+
+def test_rigid_collide(
+    one_particle_snapshot_factory,
+    simulation_factory,
+    dimer_rigid_body,
+    dimer_properties,
+):
+    snap = one_particle_snapshot_factory(particle_types=["A", "B"])
+    if snap.communicator.rank == 0:
+        snap.particles.moment_inertia[:] = [dimer_properties["inertia"]]
+        snap.particles.mass[:] = [dimer_properties["mass"][0]]
+        snap.particles.velocity[:] = [[0, 0, 1]]
+        # snap.particles.angmom[:] = [[0,0,5,0]]
+        snap.mpcd.N = 2
+        snap.mpcd.types = ["C"]
+        snap.mpcd.position[:] = [[1.1, 0, 0], [-1.1, 0, 0]]
+        snap.mpcd.velocity[:] = [[0.6, 0.7, 0.8], [0.6, -0.7, 0.8]]
+
+    sim = simulation_factory(snap)
+    sim.seed = 5
+
+    rigid = hoomd.md.constrain.Rigid()
+    rigid.body["A"] = dimer_rigid_body
+    rigid.create_bodies(sim.state)
+
+    intermed_snapshot = sim.state.get_snapshot()
+    if intermed_snapshot.communicator.rank == 0:
+        flags = (
+            intermed_snapshot.particles.typeid
+            == intermed_snapshot.particles.types.index("B")
+        )
+        intermed_snapshot.particles.mass[flags] = dimer_properties["mass"][flags]
+    sim.state.set_snapshot(intermed_snapshot)
+
+    cm = hoomd.mpcd.collide.StochasticRotationDynamics(
+        period=1,
+        angle=90,
+        embedded_particles=hoomd.filter.Rigid(flags=("constituent", "free")),
```

Comment:
Are there actually free particles in the test snapshot? It might be a good idea to have one in addition to the rigid body particles.

Reply:
Currently there are not.